src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1454,7 +1454,19 @@ private BoundExpression BindDefaultExpression(DefaultExpressionSyntax node, Bind
             TypeWithAnnotations typeWithAnnotations = this.BindType(node.Type, diagnostics, out AliasSymbol alias);
             var typeExpression = new BoundTypeExpression(node.Type, aliasOpt: alias, typeWithAnnotations);
             TypeSymbol type = typeWithAnnotations.Type;
-            return new BoundDefaultExpression(node, typeExpression, constantValueOpt: type.GetDefaultValue(), type);
+
+            bool hasErrors = false;
+
+            // Default of an enum as an attribute argument triggers serialization of the enum's type.
+            // This would fail for enums nested in a type referencing a function pointer, because
+            // function pointer serialization is not supported, see https://github.com/dotnet/roslyn/issues/48765.
+            if (this.InAttributeArgument && type.IsEnumType() && type.ContainsFunctionPointer())
+            {
+                diagnostics.Add(ErrorCode.ERR_FunctionPointerTypesInAttributeNotSupported, node.Location);
+                hasErrors = true;
```

Comment:
>hasErrors = true; [](http://example.com/codeflow?start=16&length=17)

I wouldn't set hasErrors flag due to this error. It is not a goal to record in a bound node whether any error was reported. The primary purpose of the ```HasErrors``` flag is to suppress cascading diagnostics. Typically, it is required to be set when something is "completely" wrong. The information in the node is so erroneous that anyone trying to consume the node is better to give up and stay away. Basically, if any following analysis on this this node won't make sense. Not every error leads to a situation like this. When we are overzealous, this can have a negative impact. For example, compiler might fail to perform some analysis even though there is no good reason for that, etc. In this particular situation, there is nothing wrong with the information in the node. We wouldn't be able to produce metadata for it, but we reported an error about that. The node is still good.

Reply:
I see, that makes sense, will remove it. I was following #50903 which sets `HasErrors: true`, which I guess is wrong for the same reason (although it probably doesn't make sense to change it now)?

Reply:
> it probably doesn't make sense to change it now

Up to you

Reply:
> Up to you

OK, I removed both occurrences.